### PR TITLE
【prim】change layernorm_grad rules

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR types

 Bug fixes
### PR changes

others
### Describe

Pcard-66975
layer_norm反向拆解scale ,bias 为空时bug 修复

### CODE REVIEW

File: paddle/fluid/operators/layer_norm_op.cc [hunk 1]
```diff
@@ -273,16 +273,20 @@ class LayerNormCompositeGradOpMaker : public prim::CompositeGradOpMakerBase {
...
-
     // get outputs
     paddle::Tensor x_grad = this->GetSingleInputGrad("X");
-    paddle::Tensor scale_grad = this->GetSingleInputGrad("Scale");
-    paddle::Tensor bias_grad = this->GetSingleInputGrad("Bias");
+    paddle::Tensor scale_grad;
+    paddle::Tensor bias_grad;
 
+    if (scale.is_initialized()) {
+      scale_grad = this->GetSingleInputGrad("Scale");
```
Comment (JiabinYang) [inline paddle/fluid/operators/layer_norm_op.cc]: Change logic in `GetSingleInputGrad`
```
framework::VarDesc* SingleInputGrad(const std::string& name,
                                      bool drop_empty_grad = true) const {
    auto* var = this->SingleForwardInput(name);
    if(!var){
      return nullptr;
    }
    std::string var_name = var->Name();
    auto grad_var_name = framework::GradVarName(var_name);
    if (no_grad_set_.empty() || !no_grad_set_.count(grad_var_name)) {
      (*this->grad_to_var_)[grad_var_name] = var_name;
      VLOG(8) << "Valid gradients: " << grad_var_name;
    } else {
      // TODO(jiabin): Will this cause fill zeros error?
      grad_var_name = framework::kEmptyVarName;
      if (drop_empty_grad) return nullptr;
    }

    if (original_block_->HasVar(grad_var_name)) {
      // Copy Var from original block to active block, or create a new one.
      CopyVarFromOrig(grad_var_name);
      return StaticCompositeContext::Instance().GetBlock()->FindVar(
          grad_var_name);
    } else {
      return StaticCompositeContext::Instance().GetBlock()->Var(grad_var_name);
    }
  }
```

**Review by JiabinYang — COMMENTED** [review]
one comment

File: paddle/fluid/operators/layer_norm_op.cc (hunk 1 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/operators/layer_norm_op.cc]: done

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**